### ISSUE TITLE
Validation loss is much much higher than training loss?

### ISSUE THREAD

**Issue by vedantroy** [lucidrains/imagen-pytorch#128]
As a sanity check, I logged the validation loss near the start of training (so it should roughly be equal to training loss since the model weights have barely changed). However, my validation loss is around 135 (validation batch = a few hundred images, sample values are `268,131,etc.`), whereas my training loss is 0.06.

Is there anything that stands out particularly bad about my validation loss code? I've been examining it for a while, and it seems pretty fine.

```
                trainer.eval()
                losses = []
                with torch.no_grad():
                    for batch in tqdm(val_dl):
                        imgs, tags = batch["img"].cuda(), batch["tags"]
                        loss = trainer(
                            images=imgs,
                            texts=tags,
                            unet_number=1,
                        )
                        losses.append(loss)
                print(losses)
                wandb.log(
                    {
                        "loss_validation": sum(losses) / len(losses),
                        "step": ctx["step"],
                    }
                )
                trainer.train()
```

I'm only using a single UNet.
No worries if the error isn't apparent--I'm mostly just checking to see if there's something glaring that stands out to you that I have missed.

**Comment by lucidrains** [comment]
@vedantroy i added a feature for auto random splitting the validation set from the training https://github.com/lucidrains/imagen-pytorch/commit/3175918e71a333f31630203217021edff329004f

the validation loss seems to check out on my end, so i'm not sure what's going on

**Comment by vedantroy** [comment]
My bad! The issue was I divide by 255 in the training batches, but not in the validation batches and I'm using an old version of imagen_pytorch that can accept byte inputs.
